kyo-prelude/shared/src/test/scala/kyo/ParseTest.scala

```diff
@@ -559,4 +559,207 @@ class ParseTest extends Test:
+        }
+
+        "partial token across chunks" in run {
+            val parser = Parse.int
+            val input  = Stream.init(Seq[Text]("1", "2", "34", "5"))
+            Parse.run(input)(parser).run.map { result =>
+                assert(result == Chunk(12345))
+            }
+        }
+
+        "complex token splitting" in run {
+            val parser = Parse.literal("hello world")
+            val input  = Stream.init(Seq[Text]("he", "llo", " wo", "rld"))
+            Parse.run(input)(parser).run.map { result =>
+                assert(result == Chunk(()))
+            }
+        }
+
+        "accumulated state handling" in run {
+            val parser =
+                for
+                    _ <- Parse.whitespaces
+                    n <- Parse.int
+                    _ <- Parse.whitespaces
+                    _ <- Parse.literal(",")
+                yield n
+
+            val input = Stream.init(Seq[Text]("1,", " 2 ,", "  3,"))
+            Parse.run(input)(parser).run.map { result =>
+                assert(result == Chunk(1, 2, 3))
+            }
+        }
+
+        "nested parsers with streaming" in run {
+            val numberList =
+                for
+                    _ <- Parse.char('[')
+                    n <- Parse.int
+                    _ <- Parse.char(']')
+                yield n
+
+            val input = Stream.init(Seq[Text]("[1]", "[2", "]", "[3]"))
+            Parse.run(input)(numberList).run.map { result =>
+                assert(result == Chunk(1, 2, 3))
+            }
+        }
+
+        "backtracking across chunks" in run {
+            val parser = Parse.firstOf(
+                Parse.literal("foo bar").as(1),
+                Parse.literal("foo baz").as(2)
+            )
+            val input = Stream.init(Seq[Text]("foo ", "ba", "z"))
+            Parse.run(input)(parser).run.map { result =>
+                assert(result == Chunk(2))
+            }
+        }
+
+        "with Var effect" in run {
+            val parser =
+                for
+                    r <- Parse.int
+                    _ <- Parse.attempt(Parse.char(' '))
+                    _ <- Var.update[Int](_ + 1)
+                yield r
+            val input = Stream.init(Seq[Text]("1", "2 3", "4 5"))
+            Var.runTuple(0)(Parse.run(input)(parser).run).map { (count, result) =>
+                assert(result == Chunk(12, 34, 5))
+                assert(count == 3)
+            }
+        }
+
+        "with Env effect" in run {
+            val parser =
+                for
+                    r         <- Parse.int
+                    separator <- Env.get[Char]
+                    _         <- Parse.attempt(Parse.char(separator))
+                yield r
+            val input = Stream.init(Seq[Text]("1", "2 3", "4 5"))
+            Env.run(' ')(Parse.run(input)(parser).run).map { result =>
+                assert(result == Chunk(12, 34, 5))
+            }
+        }
+    }
+
+    "ParseFailed" - {
```

Comment:
I had some trouble with SBT reporting that `ParseFailed` isn't serializable. I'm not sure why since this test shows it is.

Reply:
I've removed the test since it was making the JS build fail.